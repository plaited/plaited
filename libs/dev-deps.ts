// -- std --
export {
  assert,
  assertEquals,
  assertFalse,
  assertIsError,
  assertThrows,
} from 'https://deno.land/std@0.181.0/testing/asserts.ts'
export {
  assertSpyCall,
  assertSpyCalls,
  spy,
} from 'https://deno.land/std@0.181.0/testing/mock.ts'
export { FakeTime } from 'https://deno.land/std@0.181.0/testing/time.ts'
export { assertSnapshot } from 'https://deno.land/std@0.181.0/testing/snapshot.ts'
export { walk } from 'https://deno.land/std@0.181.0/fs/mod.ts'

// -- beautify --
export { default as beautify } from 'npm:beautify@0.0.8'

// -- esbuild --
export {
  build,
  type BuildOptions,
  type OutputFile,
  stop,
} from 'https://deno.land/x/esbuild@v0.17.12/mod.js'
export { denoPlugin } from 'https://deno.land/x/esbuild_deno_loader@0.6.0/mod.ts'
