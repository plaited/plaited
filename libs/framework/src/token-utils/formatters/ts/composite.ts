import { camelCase } from '@plaited/utils'
import { Formatter, CompositeToken } from '../../types.js'
import { hasAlias } from '../has-alias.js'
import { resolveTSVar, isValidAlias } from '../ts-utils.js'

export const composite: Formatter<CompositeToken> = (token, { tokenPath, allTokens }) => {
  const { $value } = token
  if (hasAlias($value) && !isValidAlias($value, allTokens)) {
    return ''
  }
  if (hasAlias($value)) {
    return `export const ${camelCase(tokenPath.join(' '))} = ${resolveTSVar($value, allTokens)}`
  }
  const toRet: string[] = []
  for (const key in $value) {
    const val = $value[key]
    if (isValidAlias(val, allTokens)) {
      toRet.push(`  ${key}: ${resolveTSVar(val, allTokens)},`)
    }
  }
  return [`export const ${camelCase(tokenPath.join(' '))} = {`, ...toRet, '}'].join('\n')
}
