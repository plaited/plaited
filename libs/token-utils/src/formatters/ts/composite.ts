import { camelCase } from '@plaited/utils'
import { Formatter, AliasValue, FlexToken, GridToken, TypographyToken } from '../../types.js'
import { hasAlias } from '../has-alias.js'
import { resolveTSVar } from '../ts-utils.js'

type CompositeToken = FlexToken | GridToken | TypographyToken

export const composite: Formatter<CompositeToken> = (token, { tokenPath, allTokens }) => {
  const { $value } = token
  if (hasAlias($value)) {
    return `export const ${camelCase(tokenPath.join(' '))} = ${resolveTSVar($value as AliasValue, allTokens)}`
  }
  const toRet = Object.entries($value).map(([key, val]) => `  ${key}: ${resolveTSVar(val, allTokens)},`)
  return [`export const ${camelCase(tokenPath.join(' '))} = {`, ...toRet, '}'].join('\n')
}
