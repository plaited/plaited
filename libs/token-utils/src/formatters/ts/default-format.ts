import { camelCase } from '@plaited/utils'
import { Formatter } from '../../types.js'
import { hasAlias } from '../has-alias.js'
import { isValidAlias, getCssVar } from '../ts-utils.js'

export const defaultFormat: Formatter = (token, { allTokens, tokenPath }) => {
  const { $value } = token
  if (hasAlias($value) && !isValidAlias($value, allTokens)) {
    return ''
  }
  return `export const ${camelCase(tokenPath.join(' '))} = ${getCssVar(tokenPath)}`
}
