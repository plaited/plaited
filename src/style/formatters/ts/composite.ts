import type { CompositeToken } from '../../token.types.js'
import type { Formatter } from '../formatters.types.js'
import { camelCase } from '../../../utils/case.js'
import { hasAlias } from '../has-alias.js'
import { resolveTSVar, isValidAlias } from '../ts-utils.js'

export const composite: Formatter<CompositeToken> = (token, { tokenPath, allTokens }) => {
  const { $value } = token
  if (hasAlias($value) && !isValidAlias($value, allTokens)) {
    return ''
  }
  if (hasAlias($value)) {
    return `export const ${camelCase(tokenPath.join(' '))} = ${resolveTSVar($value, allTokens)}`
  }
  const toRet: string[] = []
  for (const key in $value) {
    const val = $value[key]
    if (isValidAlias(val, allTokens)) {
      toRet.push(`  ${key}: ${resolveTSVar(val, allTokens)},`)
    }
  }
  return [`export const ${camelCase(tokenPath.join(' '))} = {`, ...toRet, '}'].join('\n')
}
