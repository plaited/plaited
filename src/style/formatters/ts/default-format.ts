import type { Formatter } from '../formatters.types.ts'
import { camelCase } from '../../../utils/case.ts'
import { hasAlias } from '../has-alias.ts'
import { isValidAlias, getCssVar } from '../ts-utils.ts'

export const defaultFormat: Formatter = (token, { allTokens, tokenPath }) => {
  const { $value } = token
  if (hasAlias($value) && !isValidAlias($value, allTokens)) {
    return ''
  }
  return `export const ${camelCase(tokenPath.join(' '))} = ${getCssVar(tokenPath)}`
}
