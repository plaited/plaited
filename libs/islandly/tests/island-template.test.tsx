import { assertSnapshot } from '../../dev-deps.ts'
import { isle, ssr } from '../mod.ts'

const Island = isle({ tag: 'z-el' }, (base) => class extends base {})
Deno.test('Island.template: shadow only', (t) => {
  assertSnapshot(
    t,
    ssr(
      <Island.template>
        <div>
          <h1>header</h1>
        </div>
      </Island.template>,
    ),
  )
})
Deno.test('Island.template: shadow and id', (t) => {
  assertSnapshot(
    t,
    ssr(
      <Island.template>
        <div id='random'>
          <h1>header</h1>
        </div>
      </Island.template>,
    ),
  )
})
Deno.test('Island.template: data-target, data-trigger, tag, shadow, and id', (t) => {
  assertSnapshot(
    t,
    ssr(
      <Island.template>
        <div
          target='random'
          dataTrigger={{
            click: 'random',
            focus: 'thing',
          }}
        >
          <h1>header</h1>
        </div>
      </Island.template>,
    ),
  )
})
Deno.test('Island.template: shadow, and mode closed', (t) => {
  assertSnapshot(
    t,
    ssr(
      <Island.template shadowRootMode='closed'>
        <div>
          <h1>header</h1>
        </div>
      </Island.template>,
    ),
  )
})
Deno.test('Island.template: shadow, and slots', (t) => {
  assertSnapshot(
    t,
    ssr(
      <Island.template>
        <div>
          <h1>header</h1>
          <slot name='slot'></slot>
        </div>
        <div slot='slot'>slotted</div>
      </Island.template>,
    ),
  )
})
Deno.test('Island.template: styles string', (t) => {
  assertSnapshot(
    t,
    ssr(
      <Island.template styles='.h1 { color: red}'>
        <div>
          <h1>header</h1>
        </div>
      </Island.template>,
    ),
  )
})
Deno.test('Island.template: styles as set ', (t) => {
  assertSnapshot(
    t,
    ssr(
      <Island.template
        styles={new Set(['.h1 { color: red}', 'div {opacity: 0.85;}'])}
      >
        <div>
          <h1>header</h1>
        </div>
      </Island.template>,
    ),
  )
})
