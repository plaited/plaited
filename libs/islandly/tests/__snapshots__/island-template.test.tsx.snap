export const snapshot = {};

snapshot[`Island.template: shadow only 1`] = `'<z-el><template shadowrootmode="open" shadowrootdelegatesfocus="true"><div><h1>header</h1></div></template></z-el>'`;

snapshot[`Island.template: shadow and id 1`] = `'<z-el><template shadowrootmode="open" shadowrootdelegatesfocus="true"><div id="random"><h1>header</h1></div></template></z-el>'`;

snapshot[`Island.template: data-target, data-trigger, tag, shadow, and id 1`] = `'<z-el><template shadowrootmode="open" shadowrootdelegatesfocus="true"><div data-trigger="click->random focus->thing" target="random"><h1>header</h1></div></template></z-el>'`;

snapshot[`Island.template: shadow, and mode closed 1`] = `'<z-el><template shadowrootmode="closed" shadowrootdelegatesfocus="true"><div><h1>header</h1></div></template></z-el>'`;

snapshot[`Island.template: shadow, and slots 1`] = `'<z-el><template shadowrootmode="open" shadowrootdelegatesfocus="true"><div><h1>header</h1> <slot name="slot"></slot></div></template> <div slot="slot">slotted</div></z-el>'`;

snapshot[`Island.template: styles string 1`] = `'<z-el><template shadowrootmode="open" shadowrootdelegatesfocus="true"><style>.h1 { color: red}</style> <div><h1>header</h1></div></template></z-el>'`;

snapshot[`Island.template: styles as set  1`] = `'<z-el><template shadowrootmode="open" shadowrootdelegatesfocus="true"><style>.h1 { color: red} div {opacity: 0.85;}</style> <div><h1>header</h1></div></template></z-el>'`;
